src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -668,8 +659,39 @@ public bool TryAppendEntireSegment(DocumentsOperationContext context, TimeSeries
             return TryPutSegmentDirectly(context, key, documentId, name, collectionName, changeVector, segment, baseline);
         }
 
+        private bool AppendEntireSegment(DocumentsOperationContext context,
+            Slice key,
+            string documentId,
+            string name,
+            CollectionName collectionName,
+            string changeVectorFromReplication,
+            TimeSeriesValuesSegment segment,
+            DateTime baseline)
+        {
+            if (IsOverlapping(context, key, collectionName, segment, baseline))
+                return false;
+
+            using (var slicer = new TimeSeriesSliceHolder(context, documentId, name, collectionName.Name).WithBaseline(baseline))
+            {
+                using (var holder = new TimeSeriesSegmentHolder(this, context, slicer, documentId, name, collectionName, changeVectorFromReplication))
+                {
+                    if (holder.LoadCurrentSegment() == false)
+                        return false;
+
+                    holder.AppendExistingSegment(segment, fromUpdateStatus: true, changeVectorFromReplication: changeVectorFromReplication);
+                    return true;
+                }
+            }
+        }
+
         public bool TryAppendEntireSegmentFromSmuggler(DocumentsOperationContext context, Slice key, CollectionName collectionName, TimeSeriesItem item)
         {
+            if (item.Segment.NumberOfLiveEntries == 0)
+            {
+                // in case of dead segment, we need to update the stats properly
+                return AppendEntireSegment(context, key, item.DocId, context.GetLazyStringForFieldWithCaching(item.Name), collectionName, item.ChangeVector, item.Segment, item.Baseline);
```

Comment:
why not to move this entirely to `TryPutSegmentDirectly`? I think we should have the logic for correctly updating the stats there.

Reply:
done